aten/src/ATen/native/MathBitsFallback.cpp

```diff
@@ -0,0 +1,119 @@
+    // tensors with conj bit set as an example, but this is also applicable for tensors with neg bit.
+    // Situations to handle:
+    //  1. Purely functional situation.  Easy: materialize all inputs and
+    //     call it a day.
+    //  2. Inplace operation.  Desugar x.add_(2) into x.conj_().add_(2).conj_().
+    //     Materialize other inputs as in (1).
+    //  3. Out-of-place operation.  Desugar add(x, 2, out=y) into y.copy_(add(x, 2))
+    //  Materialize other inputs as in (1).
+    //
+    //  It is important to be able to tell if we READ from an argument and if we
+    //  WRITE from an argument.  Conservative approach is to assume that we always
+    //  READ from an argument, but in out-of-place operations you can skip
+    //  negating inputs on entry that never get used.  In current schema we
+    //  can't easily tell if inplace situation has happened, so don't do it.
+
+    //  std::cerr << "conj fallback " << op.schema().name() << "\n";
+    const auto& arguments = op.schema().arguments();
+    const auto num_arguments = arguments.size();
+    const auto stack_start = stack->size() - num_arguments;
+
+    c10::optional<bool> is_write;
+    for (int64_t i = 0; i < num_arguments; ++i) {
+      const auto& alias_info = arguments[i].alias_info();
+      if (alias_info.has_value()) {
+        if (is_write.has_value()) {
+          TORCH_CHECK(*is_write == alias_info->isWrite(),
+            "Unsupported operator for ", op_name, " fallback: ", op.schema().name(),
+            op_name, " fallback doesn't work for operators with a mix "
+            "mutable and non-mutable inputs that alias with outputs, "
+            "this must be implemented manually.  "
+            "If you got this error on a core op, please report a bug to PyTorch.");
+        } else {
+          is_write = alias_info->isWrite();
+        }
+      }
+    }
+
+    if (is_write.has_value() && !*is_write) {
+      // We assume that view operators automatically handle the math bit
+      // correctly by propagating the dispatch key in key_set.
+      // This is not necessarily always right, so you should test these cases.
+      op.redispatchBoxed(dispatch_keys & c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, key), stack);
+      return;
+    }
+
+    // Mutable inputs to be tracked separately
+    std::vector<Tensor> mutable_inputs;
+
+    for (int64_t i = 0; i < num_arguments; ++i) {
+      auto& ivalue = (*stack)[stack_start + i];
+      if (!ivalue.isTensor()) {
+        continue;
+      }
+      const auto& argument = arguments[i];
+      bool mut_arg = false;
+      if (argument.alias_info()) {
+        // Was already tested by is_write loop above
+        TORCH_INTERNAL_ASSERT_DEBUG_ONLY(argument.alias_info()->isWrite());
+        mut_arg = true;
+      }
+      if (ivalue.isTensor()) {
+        auto* impl = ivalue.unsafeToTensorImpl();
+        if (!is_bit_set(*impl)) {
```

Comment:
yeah, thanks to @swolchok, you can get a `const Tensor&` from the ivalue at no cost, so might as well do that.

Reply:
(this wasn't true when this code was originally written)

Reply:
makes sense!